cmd/kubectl-gadget/utils/trace.go

```diff
@@ -314,6 +319,15 @@ func CreateTrace(config *TraceConfig) (string, error) {
 		return "", err
 	}
 
+	if config.TraceInitialState != "" {
+		// Once the traces are created, we wait for them to be in "Started" state,
+		// so they are ready to be used by the user.
+		_, err = waitForTraceInState(traceID, config.TraceInitialState)
+		if err != nil {
+			return "", err
+		}
```

Comment:
Mmmm I think we need to delete the trace in case of an error. Otherwise, the API is reporting an error but the trace is still there.

If you decide to handle this scenario in this way, please update the function description to explain that if `TraceInitialState` is specified, then this function will succeed only if the trace is correctly created and it goes into the requested state.